### PR TITLE
Fix linen.scan docstring example.

### LINKED ISSUE #1294: Misleading flax.linen.scan example
Below is the example provided for [`flax.linen.scan`](https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.scan.html#flax.linen.scan):

```python
class SimpleScan(nn.Module):
  @nn.compact
  def __call__(self, c, xs):
    LSTM = nn.scan(nn.LSTMCell,
                   variable_broadcast="params",
                   split_rngs={"params": False})
    return LSTM()(c, xs)

xs = random.uniform(rng_1, (batch_size, features))
carry_0 = nn.LSTMCell.initialize_carry(
    random.PRNGKey(0), (batch_size,), features)
model = SimpleScan()
variables = model.init(key_2, carry_0, xs)
out_state, out_val = model.apply(variables, carry_0, xs)
```
The default `in_axes` for `nn.scan` is `0`, so it seems like this example is scanning over the batch dimension instead of over a sequence.

I believe `xs` needs a sequence dimension in axis 0, as below:

```python
import flax
import flax.linen as nn
from jax import random

class SimpleScan(nn.Module):
  @nn.compact
  def __call__(self, c, xs):
    LSTM = nn.scan(nn.LSTMCell,
                   variable_broadcast="params",
                   split_rngs={"params": False})
    return LSTM()(c, xs)

key_1, key_2, key_3 = random.split(random.PRNGKey(0), 3)
seq_len, batch_size, features = 7, 11, 13

xs = random.uniform(key_1, (seq_len, batch_size, features))
carry_0 = nn.LSTMCell.initialize_carry(key_2, (batch_size,), features)
model = SimpleScan()
variables = model.init(key_3, carry_0, xs)
out_state, out_val = model.apply(variables, carry_0, xs)
```

Is this correct? I've read https://github.com/google/flax/discussions/1283 to verify. Thanks in advance!

### PR DESCRIPTION
**Pull request by n2cholas**
Fix linen.scan docstring example to include a sequence dimension.

Fixes #1294  (issue)

## Checklist
- [x] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other
      checks if that's the case).

Example slightly diverges from one proposed in discussion:
- Change variable names for consistency
- The input/output feature dimensions differ, to show users they don't need to be the same
- Add assertion to give readers information about output shape

### CODE REVIEW

File: flax/linen/transforms.py [hunk 1]
```diff
@@ -411,20 +411,30 @@ def scan(target: Target,
...
         return LSTM()(c, xs)
 
-    xs = random.uniform(rng_1, (batch_size, features))
-    carry_0 = nn.LSTMCell.initialize_carry(
-        random.PRNGKey(0), (batch_size,), features)
+    seq_len, batch_size, in_feat, out_feat = 20, 16, 3, 5
+    key_1, key_2, key_3 = random.split(random.PRNGKey(0), 3)
+
+    xs = random.uniform(key_1, (batch_size, seq_len, in_feat))
+    init_carry = nn.LSTMCell.initialize_carry(key_2, (batch_size,), out_feat)
```
Comment (8bitmp3) [inline flax/linen/transforms.py:432]: Nit: add a blank line here (similar to line 427) before model instantiation - this may help new users with code readability. LMKWYT.

```suggestion

    init_carry = nn.LSTMCell.initialize_carry(key_2, (batch_size,), out_feat)
```

Comment (n2cholas) [reply flax/linen/transforms.py:432]: Thanks for the suggestion, done!

**Review by jheek — APPROVED** [review]
Looks great!

**Comment by jheek** [comment]
@n2cholas could you rebase the branch please? The build was failing on master

**Comment by n2cholas** [comment]
@jheek done, it passes now!

File: flax/linen/transforms.py (hunk 1 above)
Comment (8bitmp3) [reply flax/linen/transforms.py:432]: And thank you @n2cholas 👍

**Comment by 8bitmp3** [comment]
@googlebot I consent.
